### ISSUE TITLE
`cml-runner` and `cml-send-comment` commands take ages to execute

### ISSUE THREAD

**Issue by thatGreekGuy96** [iterative/cml#973]
Hey everyone! This is something that (as far as I can tell) has started happening after the update to `0.14.1` last week. 
The `cml-runner` and `cml-send-comment` commands take a very long time to run now. 
- Up to 1h for the cml runner command 
- Up to 30 minutes for the send comment command. 

I'm attaching logs for each of these here. Not a lot of info, apart from the fact that it seems to be waiting for a really long time and then just retrying? Very weird. 

[1_Deploy Cloud Instances.txt](https://github.com/iterative/cml/files/8506514/1_Deploy.Cloud.Instances.txt)
[4_Post comment.txt](https://github.com/iterative/cml/files/8506522/4_Post.comment.txt)

**Comment by 0x2b3bfa0** [comment]
https://github.com/iterative/cml/blob/dc27724aedd23d4bdacefe6599ba4ad28a5ff949/src/drivers/github.js#L57

**Comment by 0x2b3bfa0** [comment]
This usually happens when GitHub is rate–limiting your API requests. 🤔

**Comment by 0x2b3bfa0** [comment]
Can you provide a minimal workflow to reproduce this issue?

**Comment by 0x2b3bfa0** [comment]
Also, does reverting to a previous version fix the issue? E.g. try pinning the version and replace `{VERSION}` with `0.15.0` or `0.13.0` in your workflow.

```yaml
steps:
  - uses: iterative/setup-cml@v1
    with:
      version: {VERSION}
```

**Comment by thatGreekGuy96** [comment]
Hey @0x2b3bfa0 you are right, it seems i've hit the rate limit on my account, which is very odd because I'm not submitting that many workflows. I'll have a look and let you know if this is still relevant to `cml` or not. Could there be that there is a change in how the runners behave that might cause the github API to be hit a lot more frequently than before?

**Comment by thatGreekGuy96** [comment]
@0x2b3bfa0 update on this: it looks like accross our dev/testing/prod accounts there were about 100 cml runners that had not been shut down! I guess the communication between these runners and the github api was hitting the rate limit or something? 
Next question however is why were all of these runners up in the first place to begin with....

**Comment by thatGreekGuy96** [comment]
@0x2b3bfa0 i've had a look and I'm not sure exactly what is going on but something changed after version `0.13.0` that is causing the number to make a lot of requests to the github api when they are idling and during shut down. Reverting to `0.13.0` seems to fix the issue.

I ssh-ed into the runners and had a look at the cml service logs but there was nothing I could see there that was hinting to a high volume of github api requests.

**Comment by dacbd** [comment]
@thatGreekGuy96 how were the credentials **_generated_** which were used with the `cml runner` /what cloud provider are you using?

**Comment by 0x2b3bfa0** [comment]
Sounds like a possible side effect of this change:
https://github.com/iterative/cml/issues/952#issuecomment-1097404231

**Comment by 0x2b3bfa0** [comment]
> _there were about 100 cml runners that had not been shut down!_

This is a bit more worrying. 🙀


> _I guess the communication between these runners and the github api was hitting the rate limit_

Yes, as per [my previous comment](https://github.com/iterative/cml/issues/973#issuecomment-1101889106), #948 is probably causing excessive requests.

> _Next question however is why were all of these runners up in the first place to begin with. They were all idling way past their `--idle-timeout` values._

This is not good. Maybe they didn't shut down because of the API throttling?

**Comment by dacbd** [comment]
I had issues back in December where `cml runner` instances did not self-terminate in GCP for about 2-3 weeks / and wasn't keeping a close eye out on that account. Then I made a [thing](https://github.com/dacbd/gcpcc) which I used in an action to create an issue if there were more instances than there should be in said account.

**Comment by DavidGOrtega** [comment]
This is probably introduced by the pagination feature

**Comment by 0x2b3bfa0** [comment]
@thatGreekGuy96, hopefully solved with #980, released with [`v0.15.1`](https://github.com/iterative/cml/releases/tag/v0.15.1)
